tests/integration/update.bats

```diff
@@ -586,13 +586,25 @@ EOF
 
 @test "update devices [minimal transition rules]" {
 	[[ "$ROOTLESS" -ne 0 ]] && requires rootless_cgroup
-	# This test currently only makes sense on cgroupv1.
-	requires cgroups_v1
 
-	# Run a basic shell script that tries to write to /dev/null. If "runc
-	# update" makes use of minimal transition rules, updates should not cause
-	# writes to fail at any point.
-	update_config '.process.args |= ["sh", "-c", "while true; do echo >/dev/null; done"]'
+	requires root
+
+	# Run a basic shell script that tries to read from /dev/kmsg, but
+	# due to lack of permissions, it prints the error message to /dev/null.
+	# If any data is read from /dev/kmsg, it will be printed to stdout, and the
+	# test will fail. In the same way, if access to /dev/null is denied, the
+	# error will be printed to stderr, and the test will also fail.
+	#
+	# "runc update" makes use of minimal transition rules, updates should not cause
+	# writes to fail at any point. For systemd cgroup driver on cgroup v1, the cgroup
+	# is frozen to ensure this.
+	update_config ' .linux.resources.devices = [{"allow": false, "access": "rwm"}, {"allow": false, "type": "c", "major": 1, "minor": 11, "access": "rwa"}]
+			| .linux.devices = [{"path": "/dev/kmsg", "type": "c", "major": 1, "minor": 11}]
+			| .process.capabilities.bounding += ["CAP_SYSLOG"]
+			| .process.capabilities.effective += ["CAP_SYSLOG"]
+			| .process.capabilities.inheritable += ["CAP_SYSLOG"]
+			| .process.capabilities.permitted += ["CAP_SYSLOG"]
+			| .process.args |= ["sh", "-c", "while true; do head -c 100 /dev/kmsg 2> /dev/null; done"]'
```

Comment:
@odinuge I just tried using `head -c 100 /dev/kmsg` and it fails:

```console
[kir@kir-rhat systemd]$ head -2 /dev/kmsg 
6,0,0,-;microcode: microcode updated early to revision 0xea, date = 2021-01-05
5,1,0,-;Linux version 5.12.12-300.fc34.x86_64 (mockbuild@bkernel01.iad2.fedoraproject.org) (gcc (GCC) 11.1.1 20210531 (Red Hat 11.1.1-3), GNU ld version 2.35.1-41.fc34) #1 SMP Fri Jun 18 14:30:51 UTC 2021

[kir@kir-rhat systemd]$ head -c 100 /dev/kmsg 
6,0,0,-;microcode: microcode updated early to revision 0xea, date = 2021-01-05
head: error reading '/dev/kmsg': Invalid argument

[kir@kir-rhat systemd]$ LANG=C strace head -c 100 /dev/kmsg 
execve("/usr/bin/head", ["head", "-c", "100", "/dev/kmsg"], 0x7fff61bd5808 /* 70 vars */) = 0
...
openat(AT_FDCWD, "/dev/kmsg", O_RDONLY) = 3
read(3, "6,0,0,-;microcode: microcode upd"..., 100) = 79
newfstatat(1, "", {st_mode=S_IFCHR|0620, st_rdev=makedev(0x88, 0x1), ...}, AT_EMPTY_PATH) = 0
write(1, "6,0,0,-;microcode: microcode upd"..., 796,0,0,-;microcode: microcode updated early to revision 0xea, date = 2021-01-05
) = 79
read(3, 0x7ffc5a54bfa0, 21)             = -1 EINVAL (Invalid argument)
write(2, "head: ", 6head: )                   = 6
write(2, "error reading '/dev/kmsg'", 25error reading '/dev/kmsg') = 25
write(2, ": Invalid argument", 18: Invalid argument)      = 18
write(2, "\n", 1
...
```

Seems like it happens because the kernel can only write full lines to `/dev/kmsg`.

The fix is easy: use `head -n 1` instead.

Reply:
Nah, this is OK as we're not expected to read anything from `/dev/kmsg` (if we will, it's an error).

Nevermind.

Reply:
... also, using `head -c 100` is preferable as it reads only 100 bytes, while `head -n 1` reads 8K (on my system).